lib/graphql/execution/interpreter/arguments_cache.rb

```diff
@@ -29,11 +29,16 @@ def fetch(ast_node, argument_owner, parent_object)
 
         private
 
+        NO_ARGUMENTS = {}.freeze
+
         NO_VALUE_GIVEN = Object.new
 
         def prepare_args_hash(ast_arg_or_hash_or_value)
           case ast_arg_or_hash_or_value
           when Hash
+            if ast_arg_or_hash_or_value.empty?
+              return NO_ARGUMENTS
+            end
```

Comment:
Run of the mill "use a frozen collection instead of making a new one then adding nothing to it"